src/GafferImage/Ramp.cpp

```diff
@@ -0,0 +1,280 @@
+{
+	ImageNode::affects( input, outputs );
+
+	if(
+		rampPlug()->isAncestorOf( input ) ||
+		input->parent<V2fPlug>() == positionAPlug() ||
+		input->parent<V2fPlug>() == positionBPlug() ||
+		transformPlug()->isAncestorOf( input )
+	)
+	{
+		outputs.push_back( outPlug()->channelDataPlug() );
+	}
+
+	if( formatPlug()->displayWindowPlug()->isAncestorOf( input ) )
+	{
+		outputs.push_back( outPlug()->formatPlug() );
+		outputs.push_back( outPlug()->dataWindowPlug() );
+	}
+
+	if( input == formatPlug()->pixelAspectPlug() )
+	{
+		outputs.push_back( outPlug()->formatPlug() );
+	}
+
+	if( input == layerPlug() )
+	{
+		outputs.push_back( outPlug()->channelNamesPlug() );
+	}
+}
+
+void Ramp::hashFormat( const GafferImage::ImagePlug *output, const Gaffer::Context *context, IECore::MurmurHash &h ) const
+{
+	ImageNode::hashFormat( output, context, h );
+	h.append( formatPlug()->hash() );
+}
+
+GafferImage::Format Ramp::computeFormat( const Gaffer::Context *context, const ImagePlug *parent ) const
+{
+	return formatPlug()->getValue();
+}
+
+void Ramp::hashDataWindow( const GafferImage::ImagePlug *output, const Gaffer::Context *context, IECore::MurmurHash &h ) const
+{
+	ImageNode::hashDataWindow( output, context, h );
+	h.append( formatPlug()->hash() );
+}
+
+Imath::Box2i Ramp::computeDataWindow( const Gaffer::Context *context, const ImagePlug *parent ) const
+{
+	return formatPlug()->getValue().getDisplayWindow();
+}
+
+IECore::ConstCompoundDataPtr Ramp::computeMetadata( const Gaffer::Context *context, const ImagePlug *parent ) const
+{
+	return outPlug()->metadataPlug()->defaultValue();
+}
+
+void Ramp::hashChannelNames( const GafferImage::ImagePlug *output, const Gaffer::Context *context, IECore::MurmurHash &h ) const
+{
+	ImageNode::hashChannelNames( output, context, h );
+	layerPlug()->hash( h );
+}
+
+IECore::ConstStringVectorDataPtr Ramp::computeChannelNames( const Gaffer::Context *context, const ImagePlug *parent ) const
+{
+	std::string channelNamePrefix = layerPlug()->getValue();
+	if( !channelNamePrefix.empty() )
+	{
+		channelNamePrefix += ".";
+	}
+
+	StringVectorDataPtr resultData = new StringVectorData();
+	vector<string> &result = resultData->writable();
+
+	result.push_back( channelNamePrefix + "R" );
+	result.push_back( channelNamePrefix + "G" );
+	result.push_back( channelNamePrefix + "B" );
+	result.push_back( channelNamePrefix + "A" );
+
+	return resultData;
+}
+
+void Ramp::hashChannelData( const GafferImage::ImagePlug *output, const Gaffer::Context *context, IECore::MurmurHash &h ) const
+{
+	ImageNode::hashChannelData( output, context, h );
+
+	V2i tileOrigin = context->get<V2i>( ImagePlug::tileOriginContextName );
+	h.append( tileOrigin );
+
+	string channelName = context->get<std::string>( ImagePlug::channelNameContextName );
+	h.append( channelName );
+
+	rampPlug()->hash( h );
+	transformPlug()->hash( h );
+
+	h.append( positionAPlug()->getValue() );
```

Comment:
We generally avoid calling `getValue()`, unless doing so means we can generate a more accurate hash. In this case we can't, so this should just be `positionAPlug()->hash( h )`, and the same for the end position plug too.